sycl/include/CL/sycl/detail/queue_impl.hpp

```diff
@@ -121,10 +121,12 @@ class queue_impl {
   }
 
   void throw_asynchronous() {
-    if (m_AsyncHandler && m_Exceptions.size()) {
-      m_AsyncHandler(m_Exceptions);
+    // Clear out the exception list before calling the async handler--that way,
+    // if the handler throws, we won't interfere with it in the destructor.
+    exception_list Exceptions = std::move(m_Exceptions);
```

Comment:
AFAIK it's not very safe to work with std::move'ed vector as after moving "it is in valid but unspecified state".
Anyway this was already fixed in another PR 609 for that problem, so I suggest removing this change from your patch.

Reply:
My recollection is that the move constructor of `std::vector` guarantees that the moved-from vector is empty. I can easily remove this change from the patch, but it needs #609 to land first or else tests will fail.

Reply:
Thank you for giving us some evil coding interview questions... :-)
It definitely deserves an explaining comment here...
While this makes sense and I can find note 7 of https://en.cppreference.com/w/cpp/container/vector/vector I have not been able to find a allusion to this in the latest C++ draft.
Any hint?